Pub 487

Matthew-Summerbell opened the pull request:
### JIRA link (if applicable) ###



### Change description ###
Based on HMI API Gateway testing framework.


**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

ShufenZ reviewed (commented):
build.gradle:

line 14: Need to check out version '0.0.1' is correct?
task for functional test: 
line 88 repeats line 81
line 89 repeats line 82

task for acceptance test are identical except that there are no repeated tasks as functional. So there are differences between them.
Suggestion: make the tasks in the same order as those for functional test.

task for integration has fewer tasks than functional and acceptance.
e.g.
Do we need to set assertion flag?
Do we need to set scan test classes flag?

line 128, missing sourceSets.acceptanceTest

block of definition for
integrationTestImplementation group:  
is missing but testImplementation group:

Suggestion:
Move lines 257 and 258 to between lines 246 and 248 for consistence

It seems the following is missing below line 269:
smokeTestImplementation sourceSets.test.runtimeClasspath

src/acceptanceTest/java/uk/gov/hmcts/futurehearings/pip/acceptance/common/delegate/CommonDelegateImpl.java:
Suggestion:
line 42, remove final qualifier from private static method.

src/acceptanceTest/java/uk/gov/hmcts/futurehearings/pip/acceptance/common/header/dto/factory/PayloadHeaderDTOFactory.java:
Suggestion:
line 18 and 32, remove final qualifier from static methods.

src/acceptanceTest/java/uk/gov/hmcts/futurehearings/pip/acceptance/common/helper/CommonHeaderHelper.java:
Suggestions:
line 12, remove final qualifier from static method.

src/acceptanceTest/java/uk/gov/hmcts/futurehearings/pip/acceptance/common/security/OAuthTokenGenerator.java:
redundant Exception throw clause

Suggestions:
lines 13 and 30, remove final qualifier from static methods.

src/acceptanceTest/java/uk/gov/hmcts/futurehearings/pip/acceptance/common/test/PIPCommonTest.java:

Incorrect signature for the 4 @ annotated method. Make them static.

Inconsistent styles of using switch between tow files:
src/acceptanceTest/java/uk/gov/hmcts/futurehearings/pip/acceptance/common/RestClientTemplate.java
and
src/functionalTest/java/uk/gov/hmcts/futurehearings/pip/functional/common/rest/RestClientTemplate.java

src/functionalTest/java/uk/gov/hmcts/futurehearings/pip/functional/common/security/OAuthTokenGenerator.java:
See the review comments for OAuthTokenGenerator.java for acceptanceTest

src/functionalTest/java/uk/gov/hmcts/futurehearings/pip/functional/common/test/FunctionalTest.java:
Make @BeforeAll annotated method static.

Matthew-Summerbell commented:
@ShufenZ Responses in **bold**

build.gradle:
line 14: Need to check out version '0.0.1' is correct?
**Version number is inherited from template project, Not something I setup would be best to raise with Milan and Subruto**

 task for functional test: line 88 repeats line 81 line 89 repeats line 82
**Sorted thanks**

task for acceptance test are identical except that there are no repeated tasks as functional. So there are differences between them.
**There are some minor differences, could you please explain further?**

 Suggestion: make the tasks in the same order as those for functional test.
**Done**

task for integration has fewer tasks than functional and acceptance.
e.g. Do we need to set assertion flag? Do we need to set scan test classes flag?
**This is inherited from the template, I have not added this code**

line 128, missing sourceSets.acceptanceTest
**Not included in HMI so will leave as is**

block of definition for integrationTestImplementation group: is missing but testImplementation group:
**Inherited from template. I’ve added Functional and Acceptance, not sure we will need Integration tests as not used on HMI**

Suggestion: Move lines 257 and 258 to between lines 246 and 248 for consistence
**Done**

It seems the following is missing below line 269: smokeTestImplementation sourceSets.test.runtimeClasspath
**Inherited from template I’ve added Functional and Acceptance, not smoke tests at this time.**

src/acceptanceTest/java/uk/gov/hmcts/futurehearings/pip/acceptance/common/delegate/CommonDelegateImpl.java: Suggestion: line 42, remove final qualifier from private static method.
**Done**

src/acceptanceTest/java/uk/gov/hmcts/futurehearings/pip/acceptance/common/header/dto/factory/PayloadHeaderDTOFactory.java: Suggestion: line 18 and 32, remove final qualifier from static methods.
**Done**

src/acceptanceTest/java/uk/gov/hmcts/futurehearings/pip/acceptance/common/helper/CommonHeaderHelper.java: Suggestions: line 12, remove final qualifier from static method.
**Done**

src/acceptanceTest/java/uk/gov/hmcts/futurehearings/pip/acceptance/common/security/OAuthTokenGenerator.java: redundant Exception throw clause
**Removed**

Suggestions: lines 13 and 30, remove final qualifier from static methods.
**Removed**

src/acceptanceTest/java/uk/gov/hmcts/futurehearings/pip/acceptance/common/test/PIPCommonTest.java:
Incorrect signature for the 4 @ annotated method. Make them static.
**Changed**

Inconsistent styles of using switch between tow files: src/acceptanceTest/java/uk/gov/hmcts/futurehearings/pip/acceptance/common/RestClientTemplate.java and src/functionalTest/java/uk/gov/hmcts/futurehearings/pip/functional/common/rest/RestClientTemplate.java
**Updated**

src/functionalTest/java/uk/gov/hmcts/futurehearings/pip/functional/common/security/OAuthTokenGenerator.java: See the review comments for OAuthTokenGenerator.java for acceptanceTest
**Done**

src/functionalTest/java/uk/gov/hmcts/futurehearings/pip/functional/common/test/FunctionalTest.java: Make @BeforeAll annotated method static.
**Cannot be static as not static fields are referenced**
